basics.js

```diff
@@ -4,22 +4,41 @@
 
 var pets = ['cat', 'dog', 'wombat'];
 
+for (var p in pets){
+  console.log("I have a " + pets[p]);
+};
+
 ///// Question 2
 // Change my friend's hair color to "green", without modifying the line below.
 
 var friend = { name: 'Chris', age: 27, hairColor: 'blue' };
 
+friend.hairColor = "green";
+console.log(friend);
+
 ///// Question 3
 // The function below is problematic. In 1-2 sentences explain why, and then fix the code.
 
+// it's a bit problematic because you're not really adding one inside of the function, instead returing a square then adding 1 to the return value.  Also assigning a variable
 var squarePlusOne = function(n){
-  squared = n * n;
-  return squared + 1;
+  var squared = (n * n) + 1;
+  return squared;
 };
 
+console.log(squarePlusOne(2))
 ///// Question 4
 // Translate the Ruby method defined below into a JavaScript function.
 
+var greet = function(name,excited){
+  greeting = "Hi " + name + " here!";
+  if (excited === true){
```

Comment:
No need to check if excited === true, but this still works. Just overly verbose.